Framework/API/src/AlgorithmHistory.cpp

```diff
@@ -174,6 +174,23 @@ AlgorithmHistory_sptr AlgorithmHistory::operator[](const size_t index) const {
 }
 
 /**
+* Gets the value of a specified algorithm property
+* @param name ::  The property to find
+* @returns The string value of the property
+* @throw Exception::NotFoundError if the named property is unknown
+*/
+const std::string &
+AlgorithmHistory::getPropertyValue(const std::string &name) const {
+  for (auto &hist : m_properties) {
+    if (hist->name() == name) {
+      return hist->value();
+    }
+  }
```

Comment:
For that I would need a unary fn that took the property element and compared it to the name, which I would have to pass in through a class member variable.

By the time I added that admittedly simple method, it would take more lines of code, and I think be harder to read.

http://www.cplusplus.com/reference/algorithm/find_if/